src/lib/options/validate-options.ts

```diff
@@ -0,0 +1,170 @@
+
+const validDataShapes = ['long', 'wide'];
+const validDataTypes = ['json', 'csv', 'tsv', 'xml'];
+const validLabelsPositions = ['inside', 'outside'];
+const validControlButtons = ['all', 'play', 'none'];
+const validOverlays = ['all', 'play', 'repeat', 'none'];
+const validFillDateGapsIntervals = [null, 'year', 'month', 'day'];
+const validFillDateGapsValues = ['last', 'interpolate'];
+
+export function validateOptions(options: Partial<Options>): Partial<Options> {
+  const newOptions: Partial<Options> = {};
+
+  // Validate boolean options
+  boolOpts.forEach((opt) => {
+    if (!is(options[opt], 'boolean')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate number options
+  numberOpts.forEach((opt) => {
+    if (!is(options[opt], 'number')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate string options
+  strOpts.forEach((opt) => {
+    if (!is(options[opt], 'string')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate string or number options
+  strOrNumberOpts.forEach((opt) => {
+    if (!is(options[opt], 'string') && !is(options[opt], 'number')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate function or string options
+  strOrFuncOpts.forEach((opt) => {
+    if (!is(options[opt], 'string') && !is(options[opt], 'function')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate one of options
+  if (includes(validDataShapes, options.dataShape)) {
+    newOptions.dataShape = options.dataShape;
+  }
+
+  if (includes(validDataTypes, options.dataType)) {
+    newOptions.dataType = options.dataType;
+  }
+
+  if (includes(validLabelsPositions, options.labelsPosition)) {
+    newOptions.labelsPosition = options.labelsPosition;
+  }
+
+  if (includes(validControlButtons, options.controlButtons)) {
+    newOptions.controlButtons = options.controlButtons;
+  }
+
+  if (includes(validOverlays, options.overlays)) {
+    newOptions.overlays = options.overlays;
+  }
+
+  if (includes(validFillDateGapsIntervals, options.fillDateGapsInterval)) {
+    newOptions.fillDateGapsInterval = options.fillDateGapsInterval;
+  }
+
+  if (includes(validFillDateGapsValues, options.fillDateGapsValue)) {
+    newOptions.fillDateGapsValue = options.fillDateGapsValue;
+  }
+
+  // Validate array of options
+  if (is(options.fixedOrder, 'array', 'string')) {
+    newOptions.fixedOrder = options.fixedOrder;
+  }
+
+  // Other validations
+  if (validateDataTransform(options.dataTransform)) {
+    newOptions.dataTransform = options.dataTransform;
+  }
+
+  validateColorMap(options.colorMap);
+
+  return newOptions;
+}
+
+function validateDataTransform(value: Options['dataTransform'] | undefined): boolean {
+  if (typeof value === 'undefined') return false;
+
+  if (value === null || typeof value === 'function') {
+    return true;
+  }
+
+  return false;
+}
+
+function validateColorMap(value: string[] | { [key: string]: string } | undefined): boolean {
```

Comment:
I would just simplify this to:
```ts
return is(value, 'array', 'string') || is(value, 'object');
```

Reply:
```return is(value, 'array', 'string') || is(value, 'object');```
This logic will check if value is object only but I think we need to check that every value in object is string
```return is(value, 'object') && Object.values(value).every((v) => typeof v === 'string');```

Reply:
I agree. Initially I skipped this check for simplicity. 
But I think your suggestion is better. Let's do that.